Mathlib/RingTheory/MvPolynomial/Symmetric.lean

```diff
@@ -309,4 +309,244 @@ theorem psum_isSymmetric (n : ℕ) : IsSymmetric (psum σ R n) := rename_psum _
 
 end PowerSum
 
+section Exponent
+
+open Classical
+open Composition
+open Finset
+
+variable (σ R) [CommSemiring R] [CommSemiring S] [Fintype σ] [Fintype τ]
+
+@[ext]
+structure Exponent (n : ℕ) :=
+  (exp : σ → ℕ)
+  (sum_eq : ∑ i, exp i = n)
+
+lemma Exponent.single_le_sum {n : ℕ} (s : Exponent σ n) (i : σ) : s.exp i ≤ n := by
+  have h : ∑ i, s.exp i = n := s.sum_eq
+  simp_rw [←h]
+  exact Finset.single_le_sum (fun i _ => Nat.zero_le (s.exp i)) (mem_univ i)
+  done
```

Comment:
AIUI, you can complete remove the "done" tactic. (It is used in the natural number game, but not in mathlib.)

This also applies in a number of places below.